user accounting

Sina-Ghaderi opened the pull request:
please add features to manage users and user access to resources 
example : user1 can create , delete and 2 snapshot ... container but only access to 1 gig  ram  and  cpu 20% of host 
user1 should not access to network .. profile and storage pools 
user1 should only manage his containers not other user containers 
and something like that ... 
thanks ...

nbesimi committed:
Add PyYaml library

nbesimi committed:
Prepare image for upload

nbesimi committed:
Finalize yaml creation

nbesimi committed:
Get Image properties on container export

nbesimi committed:
Export Image CLI

nbesimi committed:
Send YAML first. Add Username and Password parameters for push.

nbesimi committed:
Repalce <fingerprint>.yaml with image.yaml

nbesimi committed:
Pull Image Implementation

nbesimi committed:
Firebase Authentication. Image Import

nbesimi committed:
List HubImages

nbesimi committed:
List Hub Images

nbesimi committed:
Import Image

nbesimi committed:
Import Image - Remove file at the end

nbesimi committed:
Change IMAGE_HUB to hub.kuti.io

nbesimi committed:
Fix Modal Bug for Move,Clone,Snapshot & Export Container

ailegion committed:
Merge pull request #222 from nbesimi/feature/image-registry-yaml

Feature/image registry yaml

nbesimi committed:
Fix - Container Actions inside container-details

nbesimi committed:
Update text all container will be updated on network update

nbesimi committed:
Set the default CPU architecture for images based on LXD Host

nbesimi committed:
When no local images exists, the nightly tab should be displayed by default

nbesimi committed:
When downloading an image from Hub add the tile as alias

nbesimi committed:
Update cli - Add image group

ailegion committed:
added: Containers - show New instance instead of empty list

ailegion committed:
added: Image-details toastr and loading when downloading

nbesimi committed:
The tabs for nightly and remote should be loaded only when accessed

nbesimi committed:
Merge pull request #224 from nbesimi/fix/bug-fixes-v2.1.2

Fix/bug fixes v2.1.2

nbesimi committed:
Merge pull request #225 from ailegion/2.2-ui-updates

2.2 ui updates

nbesimi committed:
Merge pull request #226 from nbesimi/feature/nightly-remote-tabs

The tabs for nightly and remote should be loaded only when accessed

sharsllani committed:
Added publish modal mockup visualy

sharsllani committed:
Added markdown editor

nbesimi committed:
Merge pull request #227 from sharsllani/feature/hub

Feature/hub - markdown editor added

nbesimi committed:
Image Hub Integration - UI Part

nbesimi committed:
UI - Publish Image

nbesimi committed:
Integrate Hub Publish from UI

nbesimi committed:
Fix Architecture Filter. Bug Fixes

nbesimi committed:
Fix local and hub showing nan for siz

nbesimi committed:
Bump LXDUI version

nbesimi committed:
For all logos use filename: logo.png

nbesimi committed:
Merge pull request #229 from nbesimi/feature/image-hub-integration

Feature/image hub integration

sharsllani committed:
Added attribute on propertiesmessages

ailegion committed:
added: Image-details toastr and loading when downloading

sharsllani committed:
Merge branch 'develop' into fix/ui-fixes-2

nbesimi committed:
Merge pull request #232 from nbesimi/fix/bugs-v2.2

Fix Architecture Filter. Bug Fixes

nbesimi committed:
Merge pull request #231 from sharsllani/fix/ui-fixes-2

Fix/ui fixes 2

nbesimi committed:
Merge pull request #230 from ailegion/show-download-image

added: Image-details toastr and loading when downloading

nbesimi committed:
Merge pull request #233 from nbesimi/fix/loca-hub-showing-nan-size

Fix local and hub showing nan for siz

ailegion committed:
fixed: Terminal session not closing by removing shared session

nbesimi committed:
Merge pull request #234 from ailegion/terminal-session-close-fix

fixed: Terminal session not closing by removing shared session

nbesimi committed:
Implement squashfs support also

nbesimi committed:
Merge pull request #235 from nbesimi/fix/squashfs-images

Implement squashfs support also

nbesimi committed:
Finalize the publish image UI

nbesimi committed:
Merge pull request #236 from nbesimi/feature/image-hub-ui

Finalize the publish image UI

ailegion committed:
updated: Buttons toggle based on state
updated: Wizard step control manually handled

nbesimi committed:
Merge pull request #238 from ailegion/publish-form-wizard-updated

Publish form wizard update

ailegion committed:
added: Parley js library for from validation
added: Form validation on publish to hub from

nbesimi committed:
Merge pull request #239 from ailegion/publish-form-validation

Added validation to publish form

nbesimi committed:
Position authorization middle

nbesimi committed:
Hub - BETA

nbesimi committed:
Merge pull request #242 from nbesimi/fix/position-authorization-middle

Position authorization middle

nbesimi committed:
Merge pull request #243 from nbesimi/beta-hub

Hub - BETA

ailegion committed:
added: jquery ui and TagIt library
updated: tag field design

nbesimi committed:
Merge pull request #245 from ailegion/publish-image-tags

publish form tags updated

ailegion committed:
fixed: missing libraries

nbesimi committed:
Merge pull request #246 from ailegion/fix-missing-libs-tag-it

fixed: missing libraries

sharsllani committed:
Added Key value icon with description

sharsllani committed:
Merge branch 'develop' into fix/adding-key-value-desc

nbesimi committed:
Merge pull request #247 from sharsllani/fix/adding-key-value-desc

Fix/adding key value desc

nbesimi committed:
Check if file exist before upload

nbesimi committed:
Merge pull request #248 from nbesimi/fix/check-file-exist

Check if file exist before upload

nbesimi committed:
Bump flask version

nbesimi committed:
Merge pull request #249 from nbesimi/bump-flask-version

Bump flask version

nbesimi committed:
Remove preview button. Remove hardcoded data

nbesimi committed:
Merge pull request #250 from nbesimi/ui-fix-v2.2

Remove preview button. Remove hardcoded data

sharsllani committed:
Fixed - disabled the checkbox and made id checked by default

sharsllani committed:
Merge branch 'develop' into fix/checkbox-disable

nbesimi committed:
Merge pull request #251 from sharsllani/fix/checkbox-disable

Fix/checkbox disable

felix-engelmann committed:
show list of ip addresses with CIDR netmask for network interfaces

felix-engelmann committed:
main functionality with remote lxd socket, terminal not working

felix-engelmann committed:
use superclass instead of new pylxd connection

kacchan822 committed:
Fix: KeyError: 'image.os' when container hasn't 'image.os'

kacchan822 committed:
Change: refactoring LXCNetwork with lxd api client.

bedircaushi committed:
fixed dropdown menu bug

abdulla98 committed:
Changed: the legend text-alignment from left to center

Roni98 committed:
fixed: network interface table

bedircaushi committed:
fixed: changed button on container details

abdulla98 committed:
Change: Indicator UP text color

Roni98 committed:
fixed: adding a title for port mapping

jasir99 committed:
Border from 'Create' button removed

jasir99 committed:
Add Network Interface - popover fixed

vhajdari committed:
Merge pull request #256 from kacchan822/fix/findShellTypeOfContainer

Fix: KeyError: 'image.os' when container hasn't 'image.os'

vhajdari committed:
Merge pull request #258 from kacchan822/feature/LXCNetwork-with-api

Change: refactoring LXCNetwork with lxd api client.

vhajdari committed:
Merge pull request #274 from Roni98/fix/adding_title

Adding title for port mapping.

vhajdari committed:
Merge pull request #272 from abdulla98/fix/UP-indicator-text-color

Fix: Changed UP indicator text color

vhajdari committed:
Merge pull request #271 from bedircaushi/fix/buttonsChanging

Fixed: changed buttons in container details

felix-engelmann committed:
reformatted network view to fit in one line and calculate prefixes

felix-engelmann committed:
merged with upstream to resolve conflict

agniramadani committed:
Added: changed button logout bg on hover

stiltr committed:
Move from flask_jwt to flask_jwt_extended and secure UI.

vhajdari commented:
Hi @khokooli, 

RBAC would be a great addition to the project, thanks for taking on this feature. I don't see any of your commits on this PR though!  Is there any code that you're wanting to merge? Will need to review your code first before adding the features you are developing.

Please make sure to check in your code against the develop branch and open a PR for that.  Only code from the **develop** branch will be merged to **master**.

stiltr committed:
Merge remote-tracking branch 'upstream/develop' into security-fix

stiltr committed:
Apply security fixes to new content from upstream/develop

cocoy committed:
Fixed Submit form for Clone,Snapshot,Move Export Buttons.

ASolomatin committed:
Add an option to change the listening host of the application

vhajdari committed:
Merge pull request #290 from ASolomatin/allow_change_host

Add an option to change the listening host of the application

vhajdari committed:
Merge pull request #289 from cocoy/develop

Fixed Submit form for Clone,Snapshot,Move Export Buttons.

vhajdari committed:
Merge pull request #278 from agniramadani/Add/logoutButtonHoverChanged

Added: changed button logout bg on hover

stiltr committed:
Merge branch 'develop' into security-fix

ajdini committed:
Merge branch 'master' into develop

# Conflicts:
#	requirements.txt
#	setup.py

ajdini committed:
updated: Setup.py to read from requirements.txt to prevent misconfiguration

ailegion committed:
Merge pull request #283 from stiltr/security-fix

Secure UI

ajdini committed:
Merge branch 'develop' into feature/setup_read_from_requirements

# Conflicts:
#	setup.py

nbesimi committed:
Merge pull request #301 from ailegion/feature/setup_read_from_requirements

Single point of requirement config

ailegion committed:
Merge branch 'develop' into fix/popover

ailegion committed:
Merge pull request #277 from jasir99/fix/popover

Add Network Interface - popover content fixed

mineibrm committed:
Update cli.py (#312)

fixed #302

mineibrm committed:
Redundant move (#313)

* Update cli.py

fixed #302

* Update LXCContainer.py

Deleted redundant move

iyushaw commented:
Please is there a plan to include a user permissions. Thus a user can only maybe manage only assigned containers and not anymore. 

Thanks

ye-yng committed:
Added PAM authentication.

strugee committed:
Do not allow specifying passwords in CLI args

Fixes #341

strugee committed:
Fix --username not being required

Fixes #339

dependabot[bot] committed:
Bump pyyaml from 3.11 to 5.4 (#351)

Bumps [pyyaml](https://github.com/yaml/pyyaml) from 3.11 to 5.4.
- [Release notes](https://github.com/yaml/pyyaml/releases)
- [Changelog](https://github.com/yaml/pyyaml/blob/master/CHANGES)
- [Commits](https://github.com/yaml/pyyaml/compare/3.11...5.4)

---
updated-dependencies:
- dependency-name: pyyaml
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

strugee committed:
Use bcrypt for passwords (#346)

SHA1 passwords are recognized and automatically migrated.

It would be better to automatically bcrypt the existing SHA hashes, but
it would be more complicated to do that on startup and handle all the
possible login scenarios.

Fixes #326

Co-authored-by: Ajdin Idrizi <ajdin.idrizi@gmail.com>

ye-yng committed:
Fix for JWT authentication login loop. (#335)

ye-yng committed:
LXDUI VM Support (#336)

* show list of ip addresses with CIDR netmask for network interfaces

* main functionality with remote lxd socket, terminal not working

* use superclass instead of new pylxd connection

* reformatted network view to fit in one line and calculate prefixes

* Added VM visualisation to UI.

* Updated API calls. Added Image type column in the images page.

* Fixed variable network interface name.

* Changed sslverify to false when none is chosen.

* Updated pylxd api version.

* Added storage pool selection on instance creation.

Co-authored-by: Felix Engelmann <fe-github@nlogn.org>

ye-yng committed:
LXDUI Remote server terminal fix (#338)

* show list of ip addresses with CIDR netmask for network interfaces

* main functionality with remote lxd socket, terminal not working

* use superclass instead of new pylxd connection

* reformatted network view to fit in one line and calculate prefixes

* Added remote name options in metadata.

Co-authored-by: Felix Engelmann <fe-github@nlogn.org>
Co-authored-by: Ajdin Idrizi <ajdin.idrizi@gmail.com>

strugee committed:
Do not allow specifying passwords in CLI args (#342)

Fixes #341

strugee committed:
Improve systemd unit (#343)

* Depend on LXD's socket unit instead of snapd

* Use systemd Wants= instead of Requires=

This allows using LXDUI inside a LXD container, passing the socket file
through from the host. Before, systemd would fail if LXD wasn't
installed because snap.lxd.daemon.unix.socket wouldn't be available to
start.

* Add systemd security directives

strugee committed:
Fix `--debug` wanting an argument (#345)

Commit:
removed the hard coded jwt secret key and replaced it with a routine to create a new secret key during each initiation

Changes to be committed:
modified:   __metadata__.py
modified:   api/utils/authentication.py

ye-yng committed:
Merge branch 'develop' into dev_pam_auth

Commit:
Fixed pyyaml version

ye-yng committed:
Revert "Added PAM authentication."

This reverts commit d8e06c12e6b5c1fdbcb9d4a386ccdf345ab3e83e.

ye-yng committed:
Disable InsecureRequestWarning when sslVerify is false

ye-yng committed:
Extract additional instance information

ye-yng committed:
Added extra instance details to interface

ye-yng committed:
Added missing IPv4 address checker

ye-yng committed:
Added limits CPU option

ye-yng committed:
Fixed function name

ye-yng committed:
CPU core setting implementation

ye-yng committed:
Added CPU count on error

ye-yng committed:
Changed CPU information extraction method

ye-yng committed:
Import subprocess lib

ye-yng committed:
Fixed limitsCPU variable name

ye-yng committed:
Changed default CPU limit set to false

ye-yng committed:
Removed test print. Fixed CPU parameters in form

jetroni committed:
fixed config set because of missing save statement

jetroni committed:
Merge branch 'develop' into cli-require-username

jetroni committed:
Merge pull request #344 from strugee/cli-require-username

Fix `--username` not being required

jetroni committed:
Merge pull request #353 from Real-XkLi/develop

removed the hard coded jwt secret key

jetroni committed:
Merge pull request #360 from jetroni/config-set-bugfix

fixed config set because of missing save statement

jetroni committed:
Merge pull request #356 from ye-yng/dev_container_details

Extra container details on main page

ye-yng committed:
Fixed repeated limitsCpu variable

jetroni committed:
Merge pull request #337 from ye-yng/dev_pam_auth

Linux PAM authentication

jetroni committed:
Merge pull request #358 from ye-yng/dev_set_limits_cpu

Enable limits.cpu setting on container creation menu

jetroni committed:
Fixed bug on container update while validating

jetroni committed:
Merge pull request #362 from jetroni/hotfix_container_update

Fixed bug on container update while validating

bet4it committed:
Update versions of libraries

jetroni committed:
Merge pull request #364 from bet4it/dep

Update versions of libraries

jetroni committed:
Bumps pyopenssl from 22.0.0 to 24.1.0. (#380)

Adds exact version 2.1.2 for Werkzeug.
